### PR TITLE
IGNITE-9675: Deadlock on Ignite:active() and stopping grid simultaneously calling

### PR DESCRIPTION
**Pull request by avplatonov**

### CODE REVIEW

File: modules/core/src/main/java/org/apache/ignite/internal/processors/task/GridTaskProcessor.java [hunk 1]
```diff
@@ -187,7 +188,16 @@ private IgniteClientDisconnectedCheckedException disconnectedError(@Nullable Ign
     /** {@inheritDoc} */
     @SuppressWarnings("TooBroadScope")
     @Override public void onKernalStop(boolean cancel) {
-        lock.writeLock();
+        while(true) {
+            try {
+                if (lock.tryWriteLock(1, TimeUnit.SECONDS))
+                    break;
+                else
+                    Thread.sleep(1000);
```
Comment (akuznetsov-gridgain) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/task/GridTaskProcessor.java]: May be it make sense to U.warn("Await on lock while node stopping") ?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/task/GridTaskProcessor.java [hunk 2]
```diff
@@ -187,7 +188,16 @@ private IgniteClientDisconnectedCheckedException disconnectedError(@Nullable Ign
...
+            } catch (InterruptedException e) {
+                U.warn(log, e);
```
Comment (akuznetsov-gridgain) [inline modules/core/src/main/java/org/apache/ignite/internal/processors/task/GridTaskProcessor.java]: May be make sense to print a warning "Await for lock on node stop was interrupted"?

File: modules/core/src/main/java/org/apache/ignite/internal/processors/task/GridTaskProcessor.java (hunk 1 above)
Comment (avplatonov) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/task/GridTaskProcessor.java]: maybe you're right
thanks

File: modules/core/src/main/java/org/apache/ignite/internal/processors/task/GridTaskProcessor.java (hunk 2 above)
Comment (avplatonov) [reply modules/core/src/main/java/org/apache/ignite/internal/processors/task/GridTaskProcessor.java]: maybe you're right
thanks

**Pull request closed without merging**